src/version.rs

```diff
@@ -1,19 +1,108 @@
+//! Version data and functions
+//!
+//! This module contains [`Version`] struct, [`parse`] function for building
+//! [`Version`] struct from string and some helper data structures and functions.
+//!
+//! # Examples
+//!
+//! Parsing `Version` from string and checking its fields:
+//!
+//! ```rust
+//! use semver_parser::version;
+//!
+//! # fn try_main() -> Result<(), String> {
+//!
+//! let version = version::parse("1.2.3-alpha1")?;
+//! assert_eq!(version.major, 1);
+//! assert_eq!(version.minor, 2);
+//! assert_eq!(version.patch, 3);
+//! let expected_pre =
+//! vec![semver_parser::version::Identifier::AlphaNumeric(String::from("alpha1"))];
+//! assert_eq!(expected_pre, version.pre);
+//!
+//! # Ok(())
+//! # }
+//! #
+//! # fn main() {
+//! #   try_main().unwrap();
+//! # }
+//! ```
+//! [`Version`]: ./struct.Version.html
+//! [`parse`]: ./fn.parse.html
+
 use std::fmt;
 use std::str::from_utf8;
 
 use recognize::*;
 
 use common::{self, numeric_identifier};
 
+/// Structure representing version data
+///
+/// `Version` struct has some public fields representing version data, like major/minor version
+/// string, patch number and vectors of prefix and build identifiers.
+///
+/// # Examples
+///
+/// Parsing `Version` from string and checking its fields:
+///
+/// ```rust
+/// use semver_parser::version;
+///
+/// # fn try_main() -> Result<(), String> {
+///
+/// let version = version::parse("0.1.2-alpha1")?;
+/// assert_eq!(version.major, 0);
+/// assert_eq!(version.minor, 1);
+/// assert_eq!(version.patch, 2);
+/// let expected_pre = vec![version::Identifier::AlphaNumeric(String::from("alpha1"))];
+/// assert_eq!(expected_pre, version.pre);
+/// # Ok(())
+/// # }
+/// #
+/// # fn main() {
+/// #   try_main().unwrap();
+/// # }
+/// ```
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Version {
+    /// major version as number (`0` in `"0.1.2"`);
```

Comment:
Let's capitalise the _major_, _minor_ etc on these docs and use a `.` instead of a `;` at the end.